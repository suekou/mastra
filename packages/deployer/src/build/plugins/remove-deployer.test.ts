import { describe, it, expect } from 'vitest';
import { removeDeployer } from './remove-deployer';
import { fileURLToPath } from 'node:url';
import { dirname, join } from 'node:path';
import { rollup } from 'rollup';
import esbuild from 'rollup-plugin-esbuild';

describe('Remove deployer', () => {
  const _dirname = dirname(fileURLToPath(import.meta.url));

  it.for([
    ['./__fixtures__/basic.js'],
    ['./__fixtures__/basic-with-const.js'],
    ['./__fixtures__/basic-with-import.js'],
    ['./__fixtures__/basic-with-function.js'],
  ])('should remove the deployer from %s', async ([fileName]) => {
    const file = join(_dirname, fileName);

    const bundle = await rollup({
      logLevel: 'silent',
      input: file,
      cache: false,
      treeshake: 'smallest',
      plugins: [
        {
          name: 'externalize-all',
          resolveId(id) {
            return {
              id,
              external: id !== file,
            };
          },
        },
        removeDeployer(file),
        esbuild({
          target: `esnext`,
          platform: 'node',
          minify: false,
        }),
      ],
    });

    const result = await bundle.generate({
      format: 'esm',
    });

    expect(result?.output[0].code).toMatchSnapshot();
  });
});
