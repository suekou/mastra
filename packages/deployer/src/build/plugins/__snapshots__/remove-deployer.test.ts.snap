// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Remove deployer > should remove the deployer from ./__fixtures__/basic.js 1`] = `
"import { Mastra } from '@mastra/core/mastra';
import { createLogger } from '@mastra/core/logger';
import { weatherAgent } from './agents';

const mastra = new Mastra({
  agents: {
    weatherAgent
  },
  logger: createLogger({
    name: "Mastra",
    level: "info"
  })
});

export { mastra };
"
`;

exports[`Remove deployer > should remove the deployer from ./__fixtures__/basic-with-const.js 1`] = `
"import { Mastra } from '@mastra/core/mastra';
import { createLogger } from '@mastra/core/logger';
import { weatherAgent } from './agents';

const mastra = new Mastra({
  agents: {
    weatherAgent
  },
  logger: createLogger({
    name: "Mastra",
    level: "info"
  })
});

export { mastra };
"
`;

exports[`Remove deployer > should remove the deployer from ./__fixtures__/basic-with-function.js 1`] = `
"import { Mastra } from '@mastra/core/mastra';
import { createLogger } from '@mastra/core/logger';
import { weatherAgent } from './agents';

const mastra = new Mastra({
  agents: {
    weatherAgent
  },
  logger: createLogger({
    name: "Mastra",
    level: "info"
  })
});

export { mastra };
"
`;

exports[`Remove deployer > should remove the deployer from ./__fixtures__/basic-with-import.js 1`] = `
"import { Mastra } from '@mastra/core/mastra';
import { createLogger } from '@mastra/core/logger';
import { weatherAgent } from './agents';

const mastra = new Mastra({
  agents: {
    weatherAgent
  },
  logger: createLogger({
    name: "Mastra",
    level: "info"
  })
});

export { mastra };
"
`;
